staging/src/k8s.io/legacy-cloud-providers/azure/clients/vmssclient/azure_vmssclient_test.go

```diff
@@ -119,6 +120,38 @@ func TestCreateOrUpdate(t *testing.T) {
 	assert.Nil(t, rerr)
 }
 
+func TestCreateOrUpdateAsync(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	vmss := getTestVMSS("vmss1")
+	armClient := mockarmclient.NewMockInterface(ctrl)
+	future := &azure.Future{}
+
+	armClient.EXPECT().PutResourceAsync(gomock.Any(), to.String(vmss.ID), vmss).Return(future, nil).Times(1)
+
+	vmssClient := getTestVMSSClient(armClient)
+	_, rerr := vmssClient.CreateOrUpdateAsync(context.TODO(), "rg", "vmss1", vmss)
+	assert.Nil(t, rerr)
+}
+
+func TestWaitForAsyncOperationResult(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	armClient := mockarmclient.NewMockInterface(ctrl)
+	response := &http.Response{
+		StatusCode: http.StatusOK,
+		Body:       ioutil.NopCloser(bytes.NewReader([]byte(""))),
+	}
+
+	armClient.EXPECT().WaitForAsyncOperationResult(gomock.Any(), &azure.Future{}, "VMSSWaitForAsyncOperationResult").Return(response, nil)
```

Comment:
could we add an error case for it?

Reply:
added an error case in `TestCreateOrUpdateAsync`. For `TestWaitForAsyncOperationResult` i think the error case would be included in the cases of arm client because `WaitForAsyncOperationResult` is a simple wrapper of `armClient.WaitForAsyncOperationResult`.